portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -32,44 +39,84 @@
 public class DataServlet extends HttpServlet {
 
   private HashMap<String, HashMap<String,Object>> information;  
-  private List<String> messages;
+
+  private String regex = "^[a-zA-Z ]+$";
+  private Pattern pattern =  Pattern.compile(regex, Pattern.CASE_INSENSITIVE);
+  private Matcher matcher = pattern.matcher(""); 
```

Comment:
Why does this have to be an instance variable and not local to a function?

Also, when do you call usePattern on this matcher?